# Trajectory_Analysis_of_Myeloid_Progenitors_scRNA

This project is focused on determining the transcriptional heterogeneity and cell lineage trajectory in myeloid progenitor cells, which were obtained by transcriptional sorting of single cells from the bone marrow. The results indicate multiple subgroups of progenitor ells with different fates. 

## Data
**Samples:** Bone marrow myeloid progenitors mRNA profiles from single cells were generated by deep sequencing of thousands of single cells, sequenced in several batches in an Illumina NextSeq machine.
**Input type:** Raw data in CellRanger output format
**Task**: Pre-processing the sequencing reads and constructing pseudotime trajectories.

## Workflow
This pipeline processes single-cell RNA-seq data to identify genes that change along pseudotime trajectories. The steps include:

1.  **Pre-processing**:
    -   Remove poor-quality cells and genes.
    -   Normalize the data to account for sequencing depth differences.
    -   Identify highly variable genes.
    -   Scale the data.
    -   Perform PCA.
2.  **Trajectory Inference**:
    -   Use UMAP to cluster and visualize the clusters.
    -   Use Slingshot to construct cell lineages and pseudotime trajectories.
3.  **Differential Expression Analysis**:
    -   Use tradeSeq to identify genes with expression changes along trajectories.
    -   Further use the output for functional enrichment analysis.

## Output

1.  **UMAP Plot**: A plot showing cell clusters with their dimensions reduced.
2.  **Pseudotime Trajectories**: Lineage curves constructed along pseudotime.
3.  **Differential Expression Results**: Significant differentially expressed genes in a .csv file.
4.  **Plots**: Plots visualizing journey of the highest differentially expressed gene along the trajectory.

